src/cascadia/TerminalApp/JsonUtils.h

```diff
@@ -300,6 +312,22 @@ namespace TerminalApp::JsonUtils
         }
     };
 
+#ifdef WINRT_Windows_UI_H
+    template<>
+    struct ConversionTrait<winrt::Windows::UI::Color> : public ConversionTrait<til::color>
+    {
+        winrt::Windows::UI::Color FromJson(const Json::Value& json) const
+        {
+            return static_cast<winrt::Windows::UI::Color>(ConversionTrait<til::color>{}.FromJson(json))
+        }
+
+        bool CanConvert(const Json::Value& json) const
+        {
+            return ConversionTrait<til::color>{}.CanConvert(json);
+        }
+    };
+#endif
+

```

Comment:
@DHowett this one isn't working properly for some reason. It compiles but fails to convert in the app. And it outright refuses to compile for the tests.

Reply:
The cool thing about JsonUtils is if it compiles you can run it and step in to see what it’s doing. Generally it’s an “if it compiles, it will do SOMETHING” library.

However: don’t inherit from the other conversion trait when you compose them like this.

Reply:
(Ignoring the fact that I forgot a semicolon above... haha)
This works when deploying WindowsTerminal because WINRT_Windows_UI_H is defined, thus letting til::color convert to a Win.UI.Color.

For some reason, this isn't happening in the tests though:
```
7>D:\terminal\src\cascadia\TerminalApp\JsonUtils.h(327,20): error C2440: 'static_cast': cannot convert from 'til::color' to 'winrt::Windows::UI::Color' (compiling source file CommandTests.cpp)
7>D:\terminal\src\cascadia\TerminalApp\JsonUtils.h(327,1): message : No constructor could take the source type, or constructor overload resolution was ambiguous (compiling source file CommandTests.cpp)
```
Both of the test projects still include Win.UI.Color though. I don't get why WINRT_Windows_UI_H is defined for JsonUtils' ConversionTraits, but apparently not for til::color :/

Reply:
Inclusion order matters for #ifdefs. Make sure Windows UI is included before JsonUtils.